versioned/tiered/rocksdb/src/main/java/org/projectnessie/versioned/rocksdb/EntityConverter.java

```diff
@@ -0,0 +1,172 @@
+          break;
+        case RocksRef.COMMITS_KEY_LIST:
+          if (!level0.getValue().getType().equals(Entity.EntityType.LIST)) {
+            throw new UnsupportedOperationException();
+          }
+          for (Entity level1 : level0.getValue().getList()) {
+            builder.addKeyMutation(entityToKeyMutation(level1));
+          }
+          break;
+        default:
+          throw new UnsupportedOperationException();
+      }
+    }
+    return builder.build();
+  }
+
+  /**
+   * This converts an Entity into Delta protobuf object.
+   * @param entity the entity to convert
+   * @return the object converted from the entity
+   */
+  public static ValueProtos.Delta entityToDelta(Entity entity) {
+    if (!entity.getType().equals(Entity.EntityType.MAP)) {
+      throw new UnsupportedOperationException();
+    }
+
+    ValueProtos.Delta.Builder builder = ValueProtos.Delta.newBuilder();
+
+    for (Map.Entry<String, Entity> level0 : entity.getMap().entrySet()) {
+      switch (level0.getKey()) {
+        case RocksRef.COMMITS_POSITION:
+          builder.setPosition((int) level0.getValue().getNumber());
+          break;
+        case RocksRef.COMMITS_OLD_ID:
+          builder.setOldId(level0.getValue().getBinary());
+          break;
+        case RocksRef.COMMITS_NEW_ID:
+          builder.setNewId(level0.getValue().getBinary());
+          break;
+        default:
+          throw new UnsupportedOperationException();
+      }
+    }
+    return builder.build();
+  }
+
+  /**
+   * This converts an Entity into KeyMutation protobuf object.
+   * @param entity the entity to convert
+   * @return the object converted from the entity
+   */
+  public static ValueProtos.KeyMutation entityToKeyMutation(Entity entity) {
+    if (!entity.getType().equals(Entity.EntityType.MAP)) {
+      throw new UnsupportedOperationException();
+    }
+
+    ValueProtos.KeyMutation.Builder builder = ValueProtos.KeyMutation.newBuilder();
+
+    for (Map.Entry<String, Entity> level0 : entity.getMap().entrySet()) {
+      switch (level0.getKey()) {
+        case RocksRef.COMMITS_KEY_ADDITION:
+          builder.setType(ValueProtos.KeyMutation.MutationType.ADDITION);
+          builder.setKey(entityToKey(level0.getValue()));
+          break;
+        case RocksRef.COMMITS_KEY_REMOVAL:
+          builder.setType(ValueProtos.KeyMutation.MutationType.REMOVAL);
+          builder.setKey(entityToKey(level0.getValue()));
+          break;
+        default:
+          throw new UnsupportedOperationException();
+      }
+    }
+    return builder.build();
+  }
+
+  /**
+   * This converts an Entity into Key protobuf object.
+   * @param entity the entity to convert
+   * @return the object converted from the entity
+   */
+  public static ValueProtos.Key entityToKey(Entity entity) {
+    if (!entity.getType().equals(Entity.EntityType.LIST)) {
+      throw new UnsupportedOperationException();
+    }
+
+    return ValueProtos.Key.newBuilder().addAllElements(
```

Comment:
Rather than collect into an intermediate list, consider using .forEach() to add as you process the stream.

Reply:
Updated everywhere that I could find.